services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/UploadWithKnownContentLengthHelper.java

```diff
@@ -196,7 +288,7 @@ public void onSubscribe(Subscription s) {
             returnFuture.whenComplete((r, t) -> {
                 if (t != null) {
                     s.cancel();
-                    if (failureActionInitiated.compareAndSet(false, true)) {
+                    if (failureActionInitiated.compareAndSet(false, true) && !isPaused) {
```

Comment:
Can we extract them to a method, something like "shouldFailRequest()` since we have this check elsewhere?